pkg/template/helm_template.go

```diff
@@ -71,8 +75,15 @@ func (t *HelmTemplate) template(dirPath string, input io.Reader) exec.CmdRunResu
 
 	args := []string{"template", name, chartPath, "--namespace", namespace, "--include-crds"}
 
+	if t.opts.KubernetesVersion != nil {
+		args = append(args, []string{"--kube-version", t.valuesFactory.additionaDownwardAPIValues.KubernetesVersion.String()}...)
```

Comment:
```suggestion
		args = append(args, []string{"--kube-version", vals.DownwardAPI().KubernetesVersion}...)
```

My suggestion here is to promote the AdditionalDownwardAPIValues to a primary concern in the Values struct.
This struct should contain all the values that you might eventually need to add to the templating phase.
Also given that the values are supposed to be all strings we should not need to do a `.String()` call when we are using this from the Values struct

Reply:
@joaopapereira I moved things back into the Values struct, lemme know if I misunderstood the ask or if it's better now. Honestly, I am fine with it either way, I guess this one is more intentional though